### DISCUSSION TITLE
How to use HtmlSanitizer for NET45?

### DISCUSSION THREAD

**Discussion by murtazahabib** [mganss/HtmlSanitizer#402] [Q&A]
I am facing issues in utilizing the package for the target framework .NET 4.5, and cannot change the target framework due to dependencies of multiple project. 
I tried by referencing the DLLs of the HtmlSanitizer by downloading it using nuget CLI. However, keep getting below error on building the project.
**The type or namespace name 'Ganss' could not be found (are you missing a using directive or an assembly reference?)**

Request you guys to please advise on how may I use this well developed package on .NET 45.

**Comment by mganss** [comment]
Unfortunately, HtmlSanitizer no longer targets net45 because AngleSharp doesn't either 😢 You'll have to revert back to an older version of HtmlSanitizer.

**Reply by murtazahabib** [reply]
Thank you dear. But could you please point me to the version (having DLLs) which supports .NET 45?

**Reply by mganss** [reply]
The NuGet packages that support net45 are no longer available at nuget.org due to https://github.com/mganss/HtmlSanitizer/security/advisories/GHSA-8j9v-h2vp-2hhv
The latest version that supported net45 was 4.0.199.

**Reply by murtazahabib** [reply]
Thank you. But this old version has an option to clean the <style> tag? Because the application is a corporate app and vulnerable to the xss is not an option.

**Reply by murtazahabib** [reply]
Also request you to share the example of utilizing the HtmlSanitizer version 4.0.

**Reply by mganss** [reply]
Yes `<style>` tag removal has always been supported. Usage hasn't changed much since that version. If unsure, look at older versions of the README here on GitHub. Or feel free to ask here if you run into issues.

**Reply by murtazahabib** [reply]
Yes, it is working as expected, and for excluding the tag or attribute, we have to write our own implementation, right?

**Comment by tiesont** [comment]
Also worth noting (which partially explains the lack of support):

<html><body>
<h2 id="releases">Releases</h2>
			<div class="has-inner-focus">

Version | Start Date | End Date
-- | -- | --
.NET Framework 4.8.1 | Aug 9, 2022 |  
.NET Framework 4.8 | Apr 18, 2019 |  
.NET Framework 4.7.2 | Apr 30, 2018 |  
.NET Framework 4.7.1 | Oct 17, 2017 |  
.NET Framework 4.7 | Apr 11, 2017 |  
.NET Framework 4.6.2 | Aug 2, 2016 | Jan 12, 2027
.NET Framework 4.6.1 | Nov 30, 2015 | Apr 26, 2022
.NET Framework 4.6 | Jul 29, 2015 | Apr 26, 2022
.NET Framework 4.5.2 | May 5, 2014 | Apr 26, 2022
.NET Framework 4.5.1 | Jan 15, 2014 | Jan 12, 2016
**.NET Framework 4.5** | Oct 9, 2012 | **Jan 12, 2016**
.NET Framework 4.0 | Apr 12, 2010 | Jan 12, 2016
.NET Framework 3.5 Service Pack 1 | Nov 19, 2007 | Jan 9, 2029
.NET Framework 3.0 | Nov 21, 2006 | Jul 12, 2011
.NET Framework 2.0 | Feb 17, 2006 | Jul 12, 2011

</div>
</body>
</html>

https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-framework

**Reply by murtazahabib** [reply]
Thank you for the useful information
